common/types.go

```diff
@@ -383,3 +383,75 @@ func inferDeprecatedFieldsMap(fields map[string]FieldMetadata) map[string]string
 
 	return fieldsMap
 }
+
+type RegistrationResult struct {
+	RegistrationRef string
+	Result          any // struct depends on the provider
+}
+
+type SubscriptionRegistrationParams struct {
+	Request any
+}
+
+type ObjectEvents struct {
+	Events []SubscriptionEventType
+	// ["create", "update", "delete"] our regular CRUD operation events
+	// we translate to provider-specific names contact.creation
+	WatchFields []string
+	// ["email", "fax"] fields to watch for an update subscription
+	PassThroughEvents []string
+	// any non CRUD operations with provider specific event names
+	// eg)  ["contact.merged"] for hubspot or ["jira_issue:restored", "jira_issue:archived"] for jira.
+}
+
+type ObjectName string
+
+type SubscribeParams struct {
+	RegistrationRef    string // optional, needed for some providers like Hubspot
+	SubscriptionEvents map[ObjectName]ObjectEvents
+	TargetURL          string // optional
+}
+
+type SubscriptionResult struct { // this corresponds to each API call.
+	RegistrationRef string
+	SubscriptionRef string
+	Result          any
+	Objects         []ObjectName
+	Events          []SubscriptionEventType
+	// ["create", "update", "delete"]
+	// our regular CRUD operation events we translate to provider-specific names contact.creation
+	UpdateFields []string
+	// ["email", "fax"]
+	PassThroughEvents []string
+	// provider specific events ["contact.merged"] for hubspot or ["jira_issue:restored", "jira_issue:archived"] for jira.
+}
+
+type SubscribeConnector interface {
```

Comment:
Can you add a line or two above each signature explaining the function? Looks good but I want to make sure interfaces especially are well-documented.